File: node-kattis/squawk.js
var readline = require('readline');

var rl = readline.createInterface({

    input: process.stdin,
    output: process.stdout
});

function Node(){

    this.neighbors = [];
    this.squawk = [0];
}

function build_nodes(n){

    while(n > 0){

        nodes.push(new Node());
        n--;
    }
}

function build_link(i, j){

    nodes[i].neighbors.push(j);
    nodes[j].neighbors.push(i);
}

function sum_neighbors_squawks(i, ti){

    let squawks = 0;

    for (const neigh of nodes[i].neighbors) {

        squawks += nodes[neigh].squawk[ti];
    }

    return squawks;
}

// *** keep failing the last test on kattis   *** //
// *** my hunch is due to ineger overflow     *** //
// *** similar code in my python folder works *** //
const nodes = [];
let nmst;
let links_completed = 0;

rl.on('line', line =>{

    if(!nmst){

        nmst = line.split(' ').map(x => parseInt(x));
        build_nodes(nmst[0]);
    } else {

        let link = line.split(' ').map(x => parseInt(x));
        build_link(link[0], link[1]);
        links_completed++;

        if(links_completed === nmst[1]){

            // infect user at time zero
            nodes[nmst[2]].squawk[0] = 1;

            // update squawk for t minutes
            for(ti = 1; ti <= nmst[3]; ti++){

                // iterate through each user and update squawk at ti
                for (let i = 0; i < nodes.length; i++) {

                    nodes[i].squawk.push(sum_neighbors_squawks(i, ti - 1));
                }
            }

            let sum = 0;

            // count all squawks at tf
            for (const node of nodes) {

                sum += node.squawk[nmst[3]];
            }

            console.log(sum);
        }
    }
});

File: squawk.py
from sys import stdin

class Node:
    def __init__(self):
        self.neighbors = []
        self.squawk = [0]

def build_nodes(n):
    while n > 0:
        nodes.append(Node())
        n -= 1

def build_link(i, j):
    nodes[i].neighbors.append(j)
    nodes[j].neighbors.append(i)

def infect_node(i):
    nodes[i].squawk[0] = 1

def update_squawks(i, t):
    squawks = 0
    for j in nodes[i].neighbors:
        squawks += nodes[j].squawk[t]
    nodes[i].squawk.append(squawks)

def spread_infection(t):
    for ti in range(1,t+1):
        for i in range(len(nodes)):
            update_squawks(i, ti - 1)

def sum_final_squawks(t):
    final_squawks = 0
    for node in nodes:
        final_squawks += node.squawk[t]
    return final_squawks

# list of all nodes
nodes = []

# read first line
nmst = [int(x) for x in stdin.readline().split(' ')]

# build n nodes
build_nodes(nmst[0])

# read and build all links
for i in range(nmst[1]):
    line = stdin.readline().split(' ')
    build_link(int(line[0]), int(line[1]))

# infect first node
infect_node(nmst[2])

# spread infection for t iterations
spread_infection(nmst[3])

# print the final squawk
print(sum_final_squawks(nmst[3]))